### ISSUE TITLE
bitrand seems to be much slower on 0.7

### ISSUE THREAD

**Issue by dpsanders** [JuliaLang/julia#26505]
I'm seeing a bad slowdown (150x) for `bitrand` from 0.6->0.7:

0.6:
```jl
julia> using BenchmarkTools; 

julia> @btime rand(Bool, 10^6);
  1.936 ms (2 allocations: 976.70 KiB)

julia> @btime bitrand(10^6);
  18.010 μs (7 allocations: 122.42 KiB)
```

0.7:
```jl
julia> using Random, BenchmarkTools;

julia> @btime rand(Bool, 10^6);
@btime bitrand(10^6);
  1.746 ms (2 allocations: 976.70 KiB)

julia> @btime bitrand(10^6);
  2.797 ms (3 allocations: 122.23 KiB)
```

**Comment by StefanKarpinski** [comment]
It's deprecated, perhaps that's why?

**Comment by KristofferC** [comment]
This previously ended up at:

```
function rand!{T<:Union{Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64, Int128}}(r::MersenneTwister, A::Array{T})
    n=length(A)
    n128 = n * sizeof(T) ÷ 16
    rand!(r, unsafe_wrap(Array, convert(Ptr{UInt128}, pointer(A)), n128))
    for i = 16*n128÷sizeof(T)+1:n
        @inbounds A[i] = rand(r, T)
    end
    A
end
```

with the `chunks` field of the BitArray.

Now it ends up at:

```
function rand!(rng::AbstractRNG, A::AbstractArray{T}, sp::Sampler) where T
    for i in eachindex(A)
        @inbounds A[i] = rand(rng, sp)
    end
    A
end
```

where `rand(rng, sp)` just generates a random Bool, completely avoiding the specialized method.

Probably introduced in https://github.com/JuliaLang/julia/pull/23964/files cc @rfourquet

**Comment by KristofferC** [comment]
Fixed by #26529
